specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+            // |CoreWebView2RuntimeChannel.Canary \| CoreWebView2RuntimeChannel.Dev \| CoreWebView2RuntimeChannel.Beta \| CoreWebView2RuntimeChannel.Stable | WebView2 Runtime -> Beta -> Dev -> Canary | Canary -> Dev -> Beta -> WebView2 Runtime |
+            // |CoreWebView2RuntimeChannel.Canary| Canary | Canary |
+            // |CoreWebView2RuntimeChannel.Beta \| CoreWebView2RuntimeChannel.Canary \| CoreWebView2RuntimeChannel.Stable | WebView2 Runtime -> Beta -> Canary | Canary -> Beta -> WebView2 Runtime |
+            //
+            // If both a `BrowserExecutableFolder` and a `RuntimeChannelPreferenceSet` are
+            // provided, the `BrowserExecutableFolder` takes precedence. The
+            // `RuntimeChannelPreferenceSet` can be overridden by the corresponding registry
+            // override `RuntimeChannelPreferenceSet` or the environment variable
+            // `WEBVIEW2_RUNTIME_CHANNEL_PREFERENCE_SET`. Set the value to a comma-separated string of
+            // integers, which map to the `CoreWebView2RuntimeChannel` values:
+            // Stable (0), Beta (1), Dev (2), and Canary (3). For example, the value "0,2"
+            // indicates that the loader should only search for Dev channel and the WebView2
+            // Runtime, using the order indicated by `ShouldReverseChannelSearchOrder`.
+            // The loader attempts to interpret each integer and treats any invalid entry as
+            // Stable channel.
+            CoreWebView2RuntimeChannel RuntimeChannelPreferenceSet { get; set; };
+
+            // The `ShouldReverseChannelSearchOrder` property is `false` by default and
+            // the WebView2 loader searches for a Runtime channel on the machine from
+            // most to least stable using the first channel found. The default search order is:
+            // WebView2 Runtime -> Beta -> Dev -> Canary. Set `ShouldReverseChannelSearchOrder`
+            // to `true` to reverse the search order so that the loader searches for a channel
+            // from least to most stable. If a `RuntimeChannelPreferenceSet` has been provided, the
+            // loader will only search for channels in the set. See `CoreWebView2RuntimeChannel`
+            // for more details on channels. This property can be overridden by the corresponding
+            // registry key `ReverseChannelSearchOrder` or the environment variable
+            // `WEBVIEW2_REVERSE_CHANNEL_SEARCH_ORDER`. Set the value to `1` to reverse the search
+            // order. See `CreateCoreWebView2EnvironmentWithOptions` for more details on
+            // overrides.
+            Boolean ShouldReverseChannelSearchOrder { get; set; };
+        }
+    }
+
+    runtimeclass CoreWebView2Environment
+    {
+        // ...
+
+        [static_name("Microsoft.Web.WebView2.Core.ICoreWebView2EnvironmentStatics")]
+        {
+            // ...
+
+            // Get the browser version info including channel name if it is not the
```

Comment:
What does an example of a string returned by this function look like? It might be worth documenting the format. 

In the example above you use a `IsBetaChannel` function. I assume that this is something the app author writes themselves? If so we would need to document the format of the string so that an app author can reliably parse it.

Reply:
Is it literally a "strstr" for "Beta"? Could that be faked out by something like "3.1.4.5 (Beta candidate)"?

Reply:
Forgot to include this in the docs, but the format should match the format of our existing APIs:
[get_BrowserVersionString](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2environment?view=webview2-1.0.1938.49#get_browserversionstring)
[GetAvailableCoreWebView2BrowserVersionString](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.1938.49#getavailablecorewebview2browserversionstring)

I guess we don't have examples in our existing docs, but the string looks something like:
"119.0.2113.0 canary"

Reply:
Please link to the other docs and make sure we have examples there or in both.